### PR TITLE
Add `--debug` flag to any `limactl` calls in `LimaBackend`

### PR DESCRIPTION
**Pull request by adamkpickering**
Depends on #991. This branch hangs off of that PR's branch (add-debug-mode), so only the commits since that branch fall under this PR. If anyone knows a better way to do PRs that depend on other PRs please let me know.

### CODE REVIEW

File: src/k8s-engine/lima.ts
Comment (mook-as) [inline src/k8s-engine/lima.ts:200]: Since this is always false via the constructor, we can simplified it to:
```suggestion
  debug = false;
```
(And drop the assignment in the constructor.)

File: src/k8s-engine/wsl.ts
Comment (mook-as) [inline src/k8s-engine/wsl.ts:138]: Same as lima.

**Review by mook-as — CHANGES_REQUESTED** [review]
Looks reasonable.  I wonder if it might make sense for lima to have a helper function that takes `string[]` and returns the same, prepended with `--debug` as appropriate.

I've only reviewed the last seven commits (the ones that look like they're not in #991); that should merge first before this one.

**Comment by jandubois** [comment]
@adamkpickering I'm going to wait until #991 has been merged and this PR has been rebased before trying to review it.

File: src/k8s-engine/lima.ts
Comment (jandubois) [inline src/k8s-engine/lima.ts:547]: I'm not a JavaScript/TypeScript developer, so I can't tell if this is idiomatic code, but to me this is less readable than using:

```typescript
if (this.debug) {
    args = ['--debug'].concat(args)
}
```

Also the name `processedArgs` rubs me the wrong way: the args are not processed in any way, they may just be prefixed with the `--debug` option.

I'll defer to @mook-as to determine if my criticism is valid, or just my personal preference.

The same argument applies to all following changes as well, that follow a similar pattern.

File: src/k8s-engine/lima.ts
Comment (jandubois) [inline src/k8s-engine/lima.ts:570]: As I wrote above, I would prefer to see a single `args` variable that is modified to contain whatever we want.

But if we are not doing this, then at least the naming should be consistent. The code above uses `args` and `processedArgs`, but here we have `rawArgs` and `args`, so `args` here has the opposite meaning from the code earlier, which is confusing.

File: src/k8s-engine/lima.ts
Comment (jandubois) [reply src/k8s-engine/lima.ts:547]: > I'll defer to @mook-as to determine if my criticism is valid, or just my personal preference.

I'll take https://github.com/rancher-sandbox/rancher-desktop/pull/1010#pullrequestreview-816334290 as Mark agreeing with me... 😸

Comment (mook-as) [reply src/k8s-engine/lima.ts:547]: See also https://github.com/rancher-sandbox/rancher-desktop/pull/1010#pullrequestreview-816334290 — I also found that bit to be weird, though I suggested a function to mutate the array instead.  Either is fine, I think, it's just that the current state is awkward.

Comment (adamkpickering) [reply src/k8s-engine/lima.ts:547]: I will change this. Believe it or not I had a misconception about what happens when you mutate/reassign variables that are passed into a function. Turns out reassignment is safe, and mutation is not - I had thought neither was. Thanks for the feedback!

Comment (adamkpickering) [reply src/k8s-engine/lima.ts:547]: Also yeah `processedArgs` is not a great name. I couldn't come up with anything better. `prefixedArgs` is better, but it looks like we don't need it anyways.

Comment (adamkpickering) [reply src/k8s-engine/lima.ts]: Also regarding the ternary operator, I believe it is idiomatic, though I don't have the javascript experience to say so authoritatively - @mook-as, perhaps you can weigh in?

**Review by jandubois — APPROVED** [review]
I have not actually tested it, but the code looks fine to me.

File: src/k8s-engine/lima.ts
Comment (jandubois) [reply src/k8s-engine/lima.ts]: > Also regarding the ternary operator, I believe it is idiomatic,

Personally I find this clearer

```javascript
if (this.debug) {
    args = ['--debug'].concat(args);
}
```

than

```javascript
args = this.debug ? ['--debug'].concat(args) : args;
```

The later to me looks like you are just trying to save some lines, but you are putting in a redundant assignment that you expect the compiler to optimize out again.

I don't care enough to ask you to change it; maybe it is a personal preference...

**Comment by jandubois** [comment]
I'm not sure if maybe this should be a separate PR, but we also must make sure that every `limactl` command we execute is written to a log file. Ideally (at least in debug mode) with the captured stdout/stderr as well.

Having `limctl failed with exit code 1` is really not enough to tell what was failing.

File: background.ts
Comment (mook-as) [inline background.ts:75]: Rebase failure — this is (mostly) duplicating a hunk that already landed.  Check lines 65 ~ 71.

File: src/k8s-engine/k8s.ts
Comment (mook-as) [inline src/k8s-engine/k8s.ts:88]:
```suggestion
   * Whether debug mode is enabled. If this is set, the implementation should
   * emit extra debug logging if possible.
```
(Make it clearer that debug mode should only affect logging, and introduce any other change in behaviour like spawning in a debugger.)

File: src/k8s-engine/wsl.ts
Comment (mook-as) [inline src/k8s-engine/wsl.ts:240]:
```suggestion
  /** Whether debug mode is enabled */
```

**Review by mook-as — CHANGES_REQUESTED** [review]

**Review by mook-as — APPROVED** [review]
Still not really happy about how we insert `--debug`, but I guess it's at least consistent now.

**Pull request merged**
